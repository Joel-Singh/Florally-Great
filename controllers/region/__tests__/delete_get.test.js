const path = require("path");
const express = require("express");
const request = require("supertest");

const formatHtml = require("../../testingUtils/formatHtml.js");

const addGeneralMiddleware = require(path.join(
  appRoot,
  "appMiddlewares",
  "addGeneralMiddleware.js"
));
const viewEngineSetup = require(path.join(
  appRoot,
  "appMiddlewares",
  "viewEngineSetup.js"
));
const Region = require(path.join(appRoot, "models", "region.js"));

const delete_get = require("../delete_get.js");

test("delete_get controller renders with regions", async () => {
  const app = configureExpressApp();
  app.use("/", delete_get);

  await addSomeRegionsToDatabase();

  const response = await request(app).get("/");
  const formattedHtml = formatHtml(response.text);
  expect(formattedHtml).toMatchSnapshot();
});

function configureExpressApp() {
  const app = express();

  viewEngineSetup(app);
  addGeneralMiddleware(app);

  return app;
}

async function addSomeRegionsToDatabase() {
  await addRegion("name 1", "desc 1", "647cd591a7047c50e71bc287");
  await addRegion("name 2", "desc 2", "647cd591a7047c50e71bc288");
  await addRegion("name 3", "desc 3", "647cd591a7047c50e71bc289");

  async function addRegion(name, description, _id) {
    const region = new Region({ name, description, _id });
    await region.save();
  }
}
