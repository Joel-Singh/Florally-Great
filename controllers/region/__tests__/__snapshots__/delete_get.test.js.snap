// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete_get controller renders with regions 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>Delete Region</title>
    <link rel="stylesheet" href="/stylesheets/reset.css" />
    <link rel="stylesheet" href="/stylesheets/all.css" />
    <link rel="stylesheet" href="/stylesheets/form.css" />
    <link rel="stylesheet" href="/stylesheets/center-text.css" />
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
  </head>
  <body>
    <nav>
      <h1>Navigation</h1>
      <a href="/">Home</a>
      <ul class="sidebar__flowers">
        <li><a href="/flowers">All flowers</a></li>
        <li><a href="/flowers/create">Create flower</a></li>
      </ul>
      <ul class="sidebar__regions">
        <li><a href="/regions">All regions</a></li>
        <li><a href="/regions/create">Create region</a></li>
        <li><a href="/regions/delete">Delete region</a></li>
      </ul>
    </nav>
    <main>
      <h1>Choose Region to delete</h1>
      <form method="POST" action="">
        <label class="center-text" for="name 1">name 1</label
        ><input
          type="radio"
          name="regionId"
          id="name 1"
          value="647cd591a7047c50e71bc287"
        /><label class="center-text" for="name 2">name 2</label
        ><input
          type="radio"
          name="regionId"
          id="name 2"
          value="647cd591a7047c50e71bc288"
        /><label class="center-text" for="name 3">name 3</label
        ><input
          type="radio"
          name="regionId"
          id="name 3"
          value="647cd591a7047c50e71bc289"
        /><button type="submit">Delete</button>
      </form>
    </main>
  </body>
</html>
"
`;
